x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -545,12 +545,17 @@ public RoleDescriptor parse(String name, BytesReference source, XContentType xCo
         }
 
         public RoleDescriptor parse(String name, XContentParser parser) throws IOException {
-            // validate name
-            Validation.Error validationError = Validation.Roles.validateRoleName(name, true);
-            if (validationError != null) {
-                ValidationException ve = new ValidationException();
-                ve.addValidationError(validationError.toString());
-                throw ve;
+            return parse(name, parser, true);
+        }
+
+        public RoleDescriptor parse(String name, XContentParser parser, boolean validate) throws IOException {
+            if (validate) {
```

Comment:
We want to postpone validation until we handle the rest of the request so we can merge actual results with validation errors and produce one stream of results. The boolean here provides a way to skip validation that will be done later.